### DISCUSSION TITLE
{% for value in range(start, stop, step) %} 填入变量时报错

### DISCUSSION THREAD

**Discussion by TigerCubDen** [qd-today/qd#449] [Q&A] [answered]
我希望借助该功能实现贴吧的自动签到功能，当用户关注贴吧数量较多时，需要翻页遍历所有已关注贴吧。
当我把变量填进去时，程序直接报错500错误。我猜测是因为变量的类型为string，而 range() 要求 int。
这就导致该循环语句失去实际用途。
希望作者能在该语句上实现自动类型转换，以提高可用性。

![image](https://github.com/qd-today/qd/assets/55782476/1d5836b0-edb7-42dc-b830-0f4903801d6b)
![image](https://github.com/qd-today/qd/assets/55782476/0e7ce2e8-87e3-42f0-9d0f-160470981c53)

**Comment by a76yyyy** [comment, marked as answer]
duplicate #434 

框架本身支持int函数
